src/main/java/com/yahoo/bullet/parsing/LeafExpression.java

```diff
@@ -37,6 +39,15 @@ public LeafExpression() {
         value = null;
     }
 
+    @Override
+    public Set<String> getRequiredFields() {
+        Set<String> result = new HashSet<>();
+        if (value.getKind() == Value.Kind.FIELD) {
```

Comment:
return value.getKind() == FIELD ? Collections.singletonSet(value.getValue()) : Collections.emptySet();

Reply:
Note: this is ok as long as you create a new set in BinaryExpression from the resulting left and right's required sets

Reply:
Done.